### PR TITLE
Add Text Component README.md

### LINKED ISSUE #916: Create Text component Readme.md
**Is your feature request related to a problem? Please describe.**
We currently don't have explanations of what each component does, how it should be used, how it shouldn't be used.

We should add these explanations for the Text component - including the Horizontal Rule logic.

This is a prerequisite to moving components to an external component library. 

**Describe the solution you'd like**
We should have a `README.md` file in the `src/app/components/Text/` directory. 
It should have these sections.

```
- ## Description (high-level summary that will be accessible to product/UX/Test, plus link to Storybook, explain the different cases)
- ## When to use this component (including where it is currently used)
- ## When to not use this component (if this is not known, can leave the section empty)
- ## Accessibility notes (using accessiblity notes on Zeplin https://app.zeplin.io/project/5baa564c05c2eba0d2b89568/screen/5babb3dea83a3937890e374c, ensuring they make sense to non-technical audiences)
- ## Roadmap (what we have planned for this component)
- ## Additional notes (link to any relevant ADRs)
```

**Describe alternatives you've considered**
We have considered other sections in the readme, but these seem like a good starting point. We can iterate/add/change sections in future, if needed. 

**Testing notes**
The `README.md` file should be clear, understandable, to people who are not developers and give a good explanation about the scope of a component. 

**Additional context**
N/A

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by dr3**
Resolves #916

Adds readme for the Text Component

- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Review by radiocontrolled — DISMISSED** [review]

File: src/app/components/Text/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Text [![Storybook](https://github.com/storybooks/press/blob/master/badges/storybook.svg)](https://simorghstorybook.now.sh/?selectedKind=Text)
+
+## Description
+
+The 'Text' component is intended to be used to parse all article story text. It takes [markdown](https://en.wikipedia.org/wiki/Markdown) as an input and converts it to the desired HTML markup. It has special treatment for inline links as they are handled differently if they point to a link inside Simorgh.
```
Comment (pjlee11) [inline src/app/components/Text/README.md]:
```suggestion
The 'Text' component is being used to parse all article story text. It takes [markdown](https://en.wikipedia.org/wiki/Markdown) as an input and converts it to the desired HTML markup. 
```

`intended` makes it sounds like this is it's only use case, not one use case of many. 

The text component isn't worried about the Simorgh inline link logic, the SPA link logic comes from the Inline Link Container and therefore shouldn't have affect on the Text component as a reusable React component

Comment (pjlee11) [reply src/app/components/Text/README.md]: You could mention that the styling of the component can be overridden by those components (EG: the Inline Link component)

**Review by pjlee11 — CHANGES_REQUESTED** [review]
Some small changes

**Review by sareh — DISMISSED** [review]
👍 Looks good!

File: src/app/components/Text/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+The 'Text' component is being used to parse all article story text. It takes [markdown](https://en.wikipedia.org/wiki/Markdown) as an input and converts it to the desired HTML markup. 
+
+## When to use this component
+
+This component is used throughout the page in visible text. It uses the [`Paragraph`](../Paragraph) styling and thus is not suitable for all text which may have different design requirements.
+
+## Accessibility notes
+
+This component has no features specifically targeted at accessibility however it does ensure that the correct HTML tags are used, such as `<i>`, `<b>` and `<p>` which should ensure that users using screen readers are aware of the desired inference.
```
Comment (ChrisBAshton) [inline src/app/components/Text/README.md]: Interesting! I was always taught to use `<strong>` and `<em>`, which is the 'semantic' way of inferring meaning on your text. `<b>` and `<i>` are [visual cues only, and have no inference](https://stackoverflow.com/questions/3059149/when-to-use-strong-and-when-to-use-b) - could you rephrase this paragraph to make that clear? (The way it's currently worded suggests that `<b>` and `<i>` provide _more_ context to screen readers when the opposite is true).

**Review by ChrisBAshton — CHANGES_REQUESTED** [review]

File: src/app/components/Text/README.md (hunk 2 above)
Comment (dr3) [reply src/app/components/Text/README.md]: Hey, i've reworded it a little, but not 100% sure what you mean, if this doesn't look right could you maybe recommend a suggestion change please?

Comment (ChrisBAshton) [reply src/app/components/Text/README.md]: Chasing my discussion with the accessibility champions - I want to make sure I understand why these decisions were made. I think you're documenting it correctly but I'm not sure we chose the right implementation in the first place 😅. Will comment back here when I've heard more.

Comment (ChrisBAshton) [reply src/app/components/Text/README.md]: Have heard back from the champions - makes sense. Editorial aren't using bold and italic for semantic meaning (i.e. it shouldn't be read out any differently to 'normal' text). It's purely a visual thing.

I'd update to this:

```suggestion
This component uses `<i>` and `<b>` (rather than `<em>` and `<strong>`) for text formatting. This is because editorial use italics and bold purely for visual reasons, e.g. as the opening paragraph of the article.

Editorial don't want to indicate that this text has strong stress emphasis or importance, hence the use of, say, [b](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/b) over [strong](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strong). Screen readers should read out all text with the same intonation regardless of how it is visually presented.
```

**Review by ChrisBAshton — APPROVED** [review]
Looks good 👍 Thanks Drew.

**Review by sareh — APPROVED** [review]
👍 Looks good!

**Review by pjlee11 — APPROVED** [review]
